### PR TITLE
fix: SQDSDK-5597 - Get ad id in background thread

### PR DESCRIPTION
**Pull request by markvdouw**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Move the getAdIdInfo to be triggered in a background thread

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - Added a log printing the current thread in which the fun was being called (which was then removed prior to commit) 
Screen Shot 2023-12-13 at 09 11 52

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://mparticle-eng.atlassian.net/browse/SQDSDKS-5867

### CODE REVIEW

File: android-core/src/main/java/com/mparticle/internal/MPUtility.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import android.os.StatFs;
 import android.provider.Settings;
 import android.telephony.TelephonyManager;
+import android.util.Log;
```
Comment (samdozor) [inline android-core/src/main/java/com/mparticle/internal/MPUtility.java:23]:
```suggestion
```

**Review by samdozor — CHANGES_REQUESTED** [review]
would you mind also annotating the getAdIdInfo method with  @workerthread (or whatever the syntax is)?

**Comment by markvdouw** [comment]
> would you mind also annotating the getAdIdInfo method with @workerthread (or whatever the syntax is)?

This is already done

**Review by markvdouw — COMMENTED** [review]
Already has the @WorkerThread annotation in line 128

**Review by samdozor — APPROVED** [review]

**Pull request merged**